src/store/auth/actions.js

```diff
@@ -0,0 +1,27 @@
+import {
+  LOGIN_REQUEST, LOGIN_FAILS, LOGIN_SUCCESS, LOGIN_SUBMIT
+} from './types'
+
+export const loginSubmit = ({ username, password }) => ({
+  type: LOGIN_SUBMIT,
+  username,
+  password
+})
+
+export const loginFails = (errorTxt, loading) => ({
```

Comment:
you don't need to pass `loading` as argument here. just set `loading` to `false` inside reducer.